iosapp/lib-ios/Sources/LibApp/IOSReducer.swift

```diff
@@ -0,0 +1,924 @@
+    case appleFamily(AppleFamily)
+    case major(Major)
+    case supervision(Supervision)
+    case authFail(AuthFail)
+    case installFail(InstallFail)
+
+    case onParentDeviceFail
+    case childIsOnboardingFail
+
+    public enum HappyPath: Equatable {
+      case hiThere
+      case timeExpectation
+      case confirmChildsDevice
+      case explainMinorOrSupervised
+      case confirmMinorDevice
+      case confirmParentIsOnboarding
+      case confirmInAppleFamily
+      case explainTwoInstallSteps
+      case explainAuthWithParentAppleAccount
+      case dontGetTrickedPreAuth
+      case explainInstallWithDevicePasscode
+      case dontGetTrickedPreInstall
+      case optOutBlockGroups
+      case promptClearCache
+      case batteryWarning
+      case clearingCache(Int)
+      case cacheCleared
+      case requestAppStoreRating
+      case doneQuit
+    }
+
+    // TODO: carefully think thru these flows, not sure if they landed correct
+    public enum AppleFamily: Equatable {
+      case explainRequiredForFiltering
+      case explainSetupFreeAndEasy
+      case howToSetupAppleFamily
+      // TODO: check this flow, how can we incorporate "it's required to install filter" idea?
+      case explainWhatIsAppleFamily
+      case checkIfInAppleFamily
+    }
+
+    public enum Major: Equatable {
+      case explainHarderButPossible
+      case askSelfOrOtherIsOnboarding
+      case askIfOtherIsParent
+      case explainFixAccountTypeEasyWay
+      case askIfOwnsMac
+      case askIfInAppleFamily
+      case explainAppleFamily
+    }
+
+    public enum Supervision: Equatable {
+      case intro
+      case explainSupervision
+      case explainNeedFriendWithMac
+      case explainRequiresEraseAndSetup
+      case instructions
+      case sorryNoOtherWay
+    }
+
+    public enum AuthFail: Equatable {
+      case invalidAccount(InvalidAccount)
+      case authConflict
+      case authCanceled
+      case restricted
+      case passcodeRequired
+      case networkError
+      case unexpected
+
+      public enum InvalidAccount: Equatable {
+        case letsFigureThisOut
+        case confirmInAppleFamily
+        case confirmIsMinor
+        case unexpected
+      }
+    }
+
+    public enum InstallFail: Equatable {
+      case permissionDenied
+      case other(FilterInstallError)
+    }
+  }
+
+  enum Screen: Equatable {
+    case launching
+    case onboarding(Onboarding)
+    case supervisionSuccessFirstLaunch
+    case running(showVendorId: Bool)
+
+    var isRunning: Bool {
+      if case .running = self { return true }
+      return false
+    }
+  }
+
+  enum MajorOnboarder: Equatable {
+    case `self`
+    case other
+  }
+}
+
+public extension IOSReducer {
+  enum Action: Equatable {
+    public enum Interactive: Equatable {
+      public enum OnboardingBtn: Equatable {
+        case primary
+        case secondary
+        case tertiary
+      }
+
+      case onboardingBtnTapped(OnboardingBtn, String)
```

Comment:
i chose to pass along the a `String` here in the payload, which is the text on the button, purely for logging purposes. it's important to me to be able to see people's paths thru the app, so we can know where they are failing or succeeding. and without some extra information, it would be hard to mentally reconstruct, because we would just log a bunch of ".secondary was tapped from the .askIfInAppleFamily screen" and i'd need to look up what the secondary button was on that screen. this should give enough natural language context to the logs to quickly know how they proceeded thru the screens.